src/libraries/System.Security.Cryptography/tests/X509Certificates/X500DistinguishedNameTests.cs

```diff
@@ -458,6 +458,75 @@ public static void CheckCachedOids()
+        [InlineData(new [] { "0.0" }, new[] { "31020500" }, "OID.0.0=#31020500")]
+        [InlineData(new [] { "2.5.4.3" }, new [] { "04023000" }, "CN=#3000")] // OCTET STRING is implicitly stripped
+        [InlineData(new [] { "2.5.4.3" }, new [] { "040404023000" }, "CN=#04023000")] // Only one OCTET STRING is stripped
+        [InlineData(new [] { "2.5.4.3" }, new [] { "0303003000" }, "CN=#0303003000")] // BIT STRING is not implicitly stripped
+        [InlineData(new [] { "2.5.4.8" }, new [] { "0500" }, "S=#0500")]
+        [InlineData(new [] { "2.5.4.8" }, new [] { "0101FF" }, "S=#0101FF")]
+        [InlineData(new [] { "2.5.4.3", "2.5.4.8" }, new [] { "0101FF", "3000" }, "CN=#0101FF, S=#3000")]
+        [InlineData(new [] { "2.5.4.3", "2.5.4.8" }, new [] { "0C02504A", "3000" }, "CN=PJ, S=#3000")]
+        [InlineData(new [] { "2.5.4.3", "2.5.4.8" }, new [] { "0C03233030", "3000" }, "CN=\"#00\", S=#3000")]
+        [SkipOnPlatform(TestPlatforms.Browser, "Browser doesn't support an X.509 PAL")]
+        public static void Format_ComponentWithNonStringContent(string[] oids, string[] attributeValues, string expected)
+        {
+            AsnWriter writer = new AsnWriter(AsnEncodingRules.DER);
+
+            using (writer.PushSequence())
+            {
+                for (int i = 0; i < oids.Length; i++)
+                {
+                    using (writer.PushSetOf())
+                    using (writer.PushSequence())
+                    {
+                        writer.WriteObjectIdentifier(oids[i]);
+                        writer.WriteEncodedValue(Convert.FromHexString(attributeValues[i]));
+                    }
+                }
+            }
+
+            X500DistinguishedName distinguishedName = new X500DistinguishedName(writer.Encode());
+            string dnString = distinguishedName.Format(false);
+            Assert.Equal(expected, dnString);
+
+            string decode = distinguishedName.Decode(X500DistinguishedNameFlags.None);
+            Assert.Equal(expected, decode);
+        }
+
+        [Fact]
+        [SkipOnPlatform(TestPlatforms.Browser, "Browser doesn't support an X.509 PAL")]
+        public static void Format_MultiValueComponentWithNonStringContent()
+        {
+            AsnWriter writer = new AsnWriter(AsnEncodingRules.DER);
+
+            using (writer.PushSequence())
+            using (writer.PushSetOf())
+            {
+                using (writer.PushSequence())
+                {
+                    writer.WriteObjectIdentifier("2.5.4.3");
+                    writer.WriteEncodedValue(new byte[] { 0x30, 0x00 });
+                }
+
+                using (writer.PushSequence())
+                {
+                    writer.WriteObjectIdentifier("2.5.4.8");
+                    writer.WriteEncodedValue(new byte[] { 0x31, 0x00 });
+                }
+            }
+
+            const string Expected = "CN=#3000 + S=#3100";
```

Comment:
Similar to single-RDN, maybe the multi-RDN test should have a "valid sandwich", either around the whole thing, or { invalid, valid, invalid, valid }, just to show that invalidity isn't contagious.